packages/backend/src/ee/clients/Slack/SlackBot.ts

```diff
@@ -378,4 +410,68 @@ export class CommercialSlackBot extends SlackBot {
             organizationUuid,
         });
     }
+
+    private static processThreadMessages(
+        messages: MessageElement[] | undefined,
+        excludeMessageTs: string,
+    ): ThreadMessageContext | undefined {
+        if (!messages || messages.length === 0) {
+            return undefined;
+        }
+
+        const threadMessages = messages
+            .filter((msg) => msg.ts !== excludeMessageTs)
+            .map((msg) => ({
+                text: msg.text || '[message]',
+                user: msg.user || 'unknown',
+                ts: msg.ts || '',
+            }));
+
+        Logger.info(
+            `Found thread context with ${threadMessages.length} previous messages`,
+        );
+
+        return threadMessages;
+    }
+
+    /**
+     * Fetches thread messages from Slack if consent is granted
+     */
+    private static async fetchThreadMessages({
+        client,
+        channelId,
+        threadTs,
+        excludeMessageTs,
+        hasConsent,
+    }: {
+        client: WebClient;
+        channelId: string;
+        threadTs: string;
+        excludeMessageTs: string;
+        hasConsent: boolean;
+    }): Promise<ThreadMessageContext | undefined> {
+        if (!hasConsent || !threadTs) {
+            return undefined;
+        }
+
+        try {
+            const threadHistory = await client.conversations.replies({
+                channel: channelId,
+                ts: threadTs,
+                limit: 100, // TODO: What should be the limit?
+            });
```

Comment:
this has a bug where it also re-saves messages from our bot:

![CleanShot 2025-06-11 at 12.12.25@2x.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/SRLEqMEevAzoFwvhnfeq/c92d87d5-ca5c-46b3-89b0-d62e0538a201.png)

![CleanShot 2025-06-11 at 12.12.49@2x.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/SRLEqMEevAzoFwvhnfeq/e8282636-ff96-4030-a91e-0f82b26fb504.png)

You can test this by going to the old ui (new ui will be broken for this)

![CleanShot 2025-06-11 at 12.13.41@2x.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/SRLEqMEevAzoFwvhnfeq/e63b52ba-5613-4a5a-8daa-87ae5866e6b0.png)

Reply:
ah! will look into this now

Reply:
Improved filtering - this is now working with "subtype" and "bot id" filtering before sending the messages - there's a clear check before calling the aiAgentService